src/main/java/com/google/api/codegen/gapic/ProtoModels.java

```diff
@@ -29,6 +29,10 @@ private ProtoModels() {}
 
   /** Gets the interfaces for the apis in the service config. */
   public static List<Interface> getInterfaces(Model model) {
+    if (model.getServiceConfig().getApisCount() == 0) {
+      // A valid service config was not given.
```

Comment:
What is this checking for? Why is returning the list of interfaces the correct behaviour in this case?

Reply:
This is checking for a valid service config. `model.getServiceConfig()` returns a non-null object even if a service config was not given. Also, `!model.getServiceConfig().equals(Service.getDefaultInstance()` even when a service config was not given (at the very least, the `config_version` field is set to 3.

One empirical observation is that empty service config will have `model.getServiceConfig().getApisCount() == 0`. Additionally, looking at the original logic that looks at the `apis` to find the interfaces, a valid service config must have non-empty `apis` in order to have a non-empty set of `Interface`s.

Reply:
That makes sense as a check for the service config. But why shouldn't we just throw an error if no service config is provided? Is it because we want to support the proto annotations? If so, then it is not very clear that this is what is going on - is it possible to move this logic "up", closer to where we make the decision about whether or not to parse the proto annotations?

Reply:
See https://github.com/googleapis/gapic-generator/pull/2551.

There is a check in the main method to ensure that a service config option is provided.

Reply:
Resolved.